test/forms/declaration/ConsigneeDetailsSpec.scala

```diff
@@ -18,18 +18,47 @@ package forms.declaration
 
 import org.scalatest.{MustMatchers, WordSpec}
 import play.api.libs.json.{JsObject, JsValue}
+import services.Countries.allCountries
+import uk.gov.hmrc.wco.dec.MetaData
 
 class ConsigneeDetailsSpec extends WordSpec with MustMatchers {
   import ConsigneeDetailsSpec._
 
   "Method toMetadataProperties" should {
     "return proper Metadata Properties" in {
-      pending
       val consigneeDetails = correctConsigneeDetails
-      val expectedMetadataProperties: Map[String, String] = Map.empty
+      val expectedMetadataProperties: Map[String, String] =
+        Map(
+          "declaration.goodsShipment.consignee.id" -> consigneeDetails.details.eori.get,
+          "declaration.goodsShipment.consignee.name" -> consigneeDetails.details.address.get.fullName,
+          "declaration.goodsShipment.consignee.address.line" -> consigneeDetails.details.address.get.addressLine,
+          "declaration.goodsShipment.consignee.address.cityName" -> consigneeDetails.details.address.get.townOrCity,
+          "declaration.goodsShipment.consignee.address.postcodeId" -> consigneeDetails.details.address.get.postCode,
+          "declaration.goodsShipment.consignee.address.countryCode" ->
+            allCountries
+              .find(country => consigneeDetails.details.address.get.country.contains(country.countryName))
+              .map(_.countryCode)
+              .getOrElse("")
+        )
 
       consigneeDetails.toMetadataProperties() must equal(expectedMetadataProperties)
     }
+
+    "map correctly" in {
+
+      val consigneeDetails = correctConsigneeDetails
+      val metadata = MetaData.fromProperties(consigneeDetails.toMetadataProperties())
+
+      metadata.declaration must be(defined)
+      metadata.declaration.get.goodsShipment must be(defined)
+      metadata.declaration.get.goodsShipment.get.consignee must be(defined)
+      metadata.declaration.get.goodsShipment.get.consignee.get.id must be(defined)
```

Comment:
I would also check the fields' values here. To be sure that I didn't mix them up.